app/vulncode_app/models.py

```diff
@@ -50,13 +50,16 @@ class VulnerabilityReference(models.Model):
     class Meta:
         unique_together = ('vulnerability', 'source', 'reference_id')
```

Comment:
Would not the URL also be part of the unique together? just a question, not sure. We can quite likely have multiple URLs for a source that refer to the same reference_id.